Makefile

```diff
@@ -139,13 +144,113 @@ help-install:
 #
 ###############################################################
 
+### Deployment ################################################
+#
+# General release steps:
+# --------------------------------------------------------------
+#  (1) commit all files and remove any untracked files
+#      `git status` should show nothing.
+#  (2) `make release-changelog` to print a suggested
+#      update to CHANGELOG.md. Replace "XXX" with your
+#      intended tag.
+#  (3) create an annotated tag (with -a or -s)
+#  (4) `make release-prep` which:
+#     - checks the tag
+#     - creates a virtualenv
+#     - builds and installs the sdist
+#  (5) `make release-verify` which:
+#     - runs tests
+#     - builds docker
+#  (6a) if everything succeeds: run `make release-upload`
+#       and execute the commands in order.
+#  (6b) if anything goes wrong, rollback the various steps:
+#     - delete on docker hub
+#     - delete local docker image
+#     - delete tag locally
+#     - make clean
+#
+release-check:
+	@if test -z "$(VERSION)"; then                    \
+		echo VERSION is not set.;                 \
+		echo Please create a git tag;             \
+		exit 100;                                 \
+	elif [[ "$(VERSION)" == *"dirty"* ]] ; then       \
+		echo VERSION is dirty.;                   \
+		echo Please commit all files and re-tag.; \
+		exit 101;                                 \
+	else                                              \
+		echo "===============================";   \
+		echo "Releasing version: $(VERSION)";     \
+		echo "===============================";   \
+	fi;
+
+release-env: release-env/bin/activate release-env/bin/make_shell
+
+release-env/bin/activate:
+	$(call create_venv, release-env)
+	release-env/bin/pip install -r REQUIREMENTS-CI.txt
+	touch release-env/bin/activate
+
+release-env/bin/make_shell:
+	echo '#!/bin/bash' > $@
+	echo 'source release-env/bin/activate' >> $@
+	echo 'bash "$$@"' >> $@
+	chmod a+x $@
+
+release-changelog:
+	@if test -n "$(VERSION)"; then                    \
+		echo VERSION is set to $(VERSION)         \
+		echo Create your changelog before tagging.\
+		exit 102;                                 \
+	fi;
+	@echo "##" "[XXX]" - $(shell sh -c "date +'%Y-%m-%d'")
+	@git log $(shell sh -c "git describe --tags --abbrev=0")..HEAD --pretty=format:"- %s"
+	@echo; cat CHANGELOG.md; echo "[XXX]: https://github.com/spacetx/starfish/releases/tag/XXX"
```

Comment:
when I redirected to CHANGELOG.md.nxt, this `echo` didn't generate a NL between the two versions.  weirdly, when I ran `make release-changelog` and dumped stdout, it did generate the NL between the two versions...

Reply:
That's a weird one. `printf` does the same. Going to use `\n\n` for the moment.